File: frontend/react/src/entities/Patient.ts
import { GenderEnum } from "./enums";

/** Shape returned by the API – always has id */
export interface Patient {
  id: number;
  full_name: string;
  /** ISO-8601 date (YYYY-MM-DD) */
  birth_date: string;
  phone_number?: string;
  gender?: GenderEnum;
  created_by?: number;
}

/** Payload sent to POST /patients (no id yet) */
export type NewPatient = Omit<Patient, "id">;
export type UpdatePatient = Omit<Patient,"id"|"created_by">

File: Backend/domain/entities/patient.py
# domain/entities/patient.py
from dataclasses import dataclass
from datetime import date
from typing import Optional
from domain.value_objects.gender import Gender
from domain.value_objects.date_of_birth import DateOfBirth

@dataclass(slots=True)
class Patient:
    id: Optional[int]
    full_name: str
    birth_date: DateOfBirth  
    phone_number: str
    gender: Gender
    created_by: Optional[int] = None
